### ISSUE TITLE
Manager make UI Slow with high latency

### ISSUE THREAD

**Issue by davedaverave** [11cafe/comfyui-workspace-manager#75]
When I install your fantastic add-on it works fine.. 
But after a week ComfyUI start becoming slower: Queue responds after a while, i need to wait a lot to see that my generation is starting after the prompt is read (that's said in the logs), high latency pheraps
Maybe it's because i'm stacking several workflow in your add-on ... 

I'm using runpod or other online services, does that make any difference?
After uninstalling your add-on it got back to work properly :(
can you help me?

**Comment by Weixuanf** [comment]
oh that weird...it shouldn't affect the Queue Prompt speed. Could you please screenshot your ComfyUI or screen recording?
> I'm using runpod or other online services, does that make any difference?
I haven't tested online service, I'm running locally. It may have some differences, I probably need to test on runpod myself

**Comment by Weixuanf** [comment]
I have just pushed some RAM performance improvement: https://github.com/11cafe/comfyui-workspace-manager/pull/77
It will reduce the RAM capacity that workspace-manager takes up

**Comment by davedaverave** [comment]
Thanks you very much, i just installed the new update... ill update you in a week! Great Job btw ❤️

**Comment by Weixuanf** [comment]
> Thanks you very much, i just installed the new update... ill update you in a week! Great Job btw ❤️

Thank you! Let me know if it works!

**Comment by davedaverave** [comment]
It happens like that: i presse queue button, it say 0, i need to wait 5-10sec or more, in the mean time console says "got prompt" then the generation it's added to queue and it start generate... the latency unfortunately is there even with the update :( (it does happen after some usage, after i gues i have some different workflow in this add-on

**Comment by Weixuanf** [comment]
how many workflows do you have in total approximately in your workspace drawer list

**Comment by Weixuanf** [comment]
so  at the beginning it's not slow, after you use comfyui for a while, it becomes slow? and you said you are using Runpod? and what browser are you using?

**Comment by davedaverave** [comment]
I have 4 workflows, two have approx 30 nodes, i'm using paperspace notebooks with A6000, and yes first generations after i start the notebook tend to have no latency... fresh install fo the add-on has no latency too

**Comment by Weixuanf** [comment]
umm sounds not a problem of number of workflows. What browser are you using? Chrome, firefox, etc.? After how long of usage, it starts to have latency, a few hours, or a few minutes?

i just tested in my machine Macbook Pro M1 64G Ram
This is an animate diff flow, so when I press "Queue Prompt", it says got prompt, then after 2-3 seconds it starts from 0%, I think this looks normal:
https://youtu.be/nf7NNoJe6lU

**Comment by davedaverave** [comment]
I use chrome.. some time in a matter of minutes it start having latency... some other times it does happen at the very start and last fore the entire time... without the addon any workflow start isntantly :(

it seems that complicated workflow takes longer...
if i dont use the notebook for like 30 minutes and go back to it, it goes even slower...

**Comment by davedaverave** [comment]
I'm using ComfyUI Paperspace Notebook From https://github.com/TheLastBen/fast-stable-diffusion
it does install some dependencies at the very start: 

```
force_reinstall= True
# Set to true only if you want to install the dependencies again.

#--------------------
with open('/dev/null', 'w') as devnull:import requests, os, importlib;open('/notebooks/mainppsComfy.py', 'wb').write(requests.get('https://huggingface.co/datasets/TheLastBen/PPS/raw/main/Scripts/mainppsComfy.py').content);os.chdir('/notebooks');import mainppsComfy;importlib.reload(mainppsComfy);from mainppsComfy import *;Deps(force_reinstall)
```

then i do this cell to update evrything:

``
!pip install -r /notebooks/ComfyUI/custom_nodes/was-node-suite-comfyui/requirements.txt
!pip install -r /notebooks/ComfyUI/custom_nodes/ComfyUI-KJNodes/requirements.txt 
!pip install --pre torch torchvision torchaudio --index-url https://download.pytorch.org/whl/nightly/cu121
!pip install xformers -U
!pip install insightface
!pip install onnxruntime -U
!pip install onnx -U
!pip install onnxruntime-gpu -U
!pip install pyOpenSSL
``

**Comment by Weixuanf** [comment]
Thanks for the very detailed paperspace instruction, I'll see if I can reproduce this. It sounds like a serious problem. Thanks for reporting it.

Could you help check the Browser Console? You can observe it when you press "Queue Prompt" see if there is any error or unusual logs, you can copy or screenshot them if there is any.
you can press F12 in keyboard or right click inspect to pull up the browser console.:
<img width="468" alt="Screenshot 2024-01-06 at 11 41 23 PM" src="https://github.com/11cafe/comfyui-workspace-manager/assets/18367033/3cbd7a76-d1be-4624-b5a6-b03a320985f9">

<img width="302" alt="Screenshot 2024-01-06 at 11 41 11 PM" src="https://github.com/11cafe/comfyui-workspace-manager/assets/18367033/843ce6b2-b5a1-47ab-b9a0-9e39fdb0a81b">

**Comment by davedaverave** [comment]
Thank you for your support ❤️
It seems to me that this message apparead in the Browser Console:        
Failed to load resource: the server responded with a status of 400 ()

At the same time i noticed when i add nosed the que doesn't start suddenly, and no messages are gound in the browser console ... like it's adding them into the manager and then into the workflow before queing.. is it possible ? (i'm not a coder, it's jsut the first thing that came to my mind..)

**Comment by Weixuanf** [comment]
Hi I used google colab to reproduced this issue (paperspace kicked out my account for some reason) I have some findings:

This is probably caused by the bundle size of our app (larger than pure js app since we’re using react) and how we’re storing out db now.

to solve this issue, I’m planning to:
1) migrate to indexdb so there won’t be so many network requests to server
2) enable dynamic chunk loading in react

it’s gonna take some time to fix this issue. I’ll update here once it’s done.

**Comment by davedaverave** [comment]
Your wonderfull! And i need to say Thank your for all the effort you put in this! THANKS 💚

**Comment by davedaverave** [comment]
Paperspace has some policy about cloudfare .. maybe you used methods that paperspace doesn't accept to have public adress :/ 
If i can be helpfull in any way just tell me !

**Comment by davedaverave** [comment]
Hey any news about the update?

**Comment by Weixuanf** [comment]
yesss i finally have it in beta! https://github.com/11cafe/comfyui-workspace-manager?tab=readme-ov-file#beta-preview

do you know how to use git? you can `git fetch origin beta && git checkout beta ` to test it
** make sure you DISABLED Auto Saving in settings **
it should be unchecked here:
<img width="502" alt="Screenshot 2024-01-22 at 2 33 00 PM" src="https://github.com/11cafe/comfyui-workspace-manager/assets/18367033/8a8cf911-57f6-46ef-a0d9-42b1511a3b1f">

this version eliminates all calls to python server when you don't click save. So for testing DO NOT click save button and see if it still impacts your performance. 
also DON'T click the models button yet when testing performance, since clicking models button it will run some python server scripts to scan model files in your comfyui disk

in summary, to test this:
1. switch to beta
2. disable auto save (uncheck the "enable auto save" in settings)
3. do not click save button
4. do not click models button
5. see if you still have the slow issue when queue prompt.
not that when loading the UI, the workspace manager may take 1 to 10 seconds to load and appear on your screen initially depending on your bandwidth, but once it's loaded you can use it smoothly. I found it loads very slow in google colab,,,

**Comment by davedaverave** [comment]
I'll try it istanly! God bless you

**Comment by davedaverave** [comment]
I did git clone -b beta https://github.com/11cafe/comfyui-workspace-manager.git is it correct?

**Comment by Weixuanf** [comment]
is it still not working 🥺

**Comment by davedaverave** [comment]
IT seeeemsthat all problem are gone 🥵🚀❤️❤️

**Comment by Weixuanf** [comment]
> IT seeeemsthat all problem are gone 🥵🚀❤️❤️

omggg I'm so glad to hear!! the improvement is working, my efforts are not in vain 🥹

as for saving, you can still save, it will just send a request to the the python server and write to the files under `comfyui-workspace-manager/db`, but it's done in another thread, so theoretically it shouldn't impact your py server performance even while writing.  I was just trying to be extra cautious while testing perf, controlling the test conditions.

once the file writing is done, it should not have any impact on the python server at all. In local the i/o is really fast, like 100ms? not sure in cloud how it will work, 

now you can play with the plugin as much as you want, theoretically it shouldn't slow you down any more. but you can observe, if any particular operations are causing any slow downs.

**Comment by davedaverave** [comment]
You did an amazing job, you're a wizard! i don't think there will be any pronlem !! of course ill update you ... It's real time! no latency, it seems there's no issues in webb ispector... now i'll try the other functions like saving etc..this makes my life so easieeeer THANKS A LOT

**Comment by Weixuanf** [comment]
🥹🥹🥹thanks for your kind words. I’m really glad it worked out for you! 🎉 hope you enjoy the plugin! 🤗🤗

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by Weixuanf** [comment]
You're probably the first to try our beta version, so would love to hear any feedbacks or issues you see!

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by Weixuanf** [comment]
> I am so lucky ahah it's Banger update, i uploaded my big workflows and nothing seem to bother me.. I will try now autosave!
> 
> btw if i could suggest something it would be to have and update tools in models, that let you check and in case update models, embedds, lroa ...🚀

haha thanks! 🤗🤗 Yeah we did think about the ability to auto check update for your models in the models drawer, but haven't get to that yet! And recently CivitAi has enforced you to login to download models, which is putting another challenge to our model installing functionality 😕😕
